pkg/clustermanager/eksa_mover.go

```diff
@@ -0,0 +1,165 @@
+		maxTime := time.Duration(math.MaxInt64)
+		a.applyClusterTimeout = maxTime
+		a.waitForFailureMessage = maxTime
+	}
+}
+
+// WithMoverApplyClusterTimeout allows to configure how long the mover retries
+// to apply the objects in case of failure.
+// Generally only used in tests.
+func WithMoverApplyClusterTimeout(timeout time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.applyClusterTimeout = timeout
+	}
+}
+
+// WithMoverRetryBackOff allows to configure how long the mover waits between requests
+// to update the cluster spec objects and check the status of the Cluster.
+// Generally only used in tests.
+func WithMoverRetryBackOff(backOff time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.retryBackOff = backOff
+	}
+}
+
+// Move applies the cluster's namespace and spec without checking for reconcile conditions.
+func (m *Mover) Move(ctx context.Context, spec *cluster.Spec, fromCluster, toCluster *types.Cluster) error {
+	m.log.V(3).Info("Moving the cluster object")
+	err := retrier.New(
+		m.applyClusterTimeout,
+		retrier.WithRetryPolicy(retrier.BackOffPolicy(m.retryBackOff)),
+	).Retry(func() error {
+		var err error
+		fromClient, err := m.clientFactory.BuildClientFromKubeconfig(fromCluster.KubeconfigFile)
+		if err != nil {
+			return errors.Wrap(err, "building client to read cluster objects")
+		}
+
+		toClient, err := m.clientFactory.BuildClientFromKubeconfig(toCluster.KubeconfigFile)
+		if err != nil {
+			return errors.Wrap(err, "building client to move cluster objects")
+		}
+
+		// read the cluster from bootstrap
+		cluster := &v1alpha1.Cluster{}
+		if err := fromClient.Get(ctx, spec.Cluster.Name, spec.Cluster.Namespace, cluster); err != nil {
+			return errors.Wrapf(err, "getting cluster from %s", fromCluster.Name)
+		}
+
+		// pause cluster on bootstrap
+		cluster.PauseReconcile()
+		if err := fromClient.ApplyServerSide(ctx, defaultFieldManager, cluster, kubernetes.ApplyServerSideOptions{ForceOwnership: true}); err != nil {
+			return errors.Wrapf(err, "getting cluster from %s", fromCluster.Name)
+		}
+
+		if err := moveClusterResource(ctx, cluster, toClient); err != nil {
+			return err
+		}
+
+		if err := moveChildObjects(ctx, spec, fromClient, toClient); err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+func moveClusterResource(ctx context.Context, cluster *v1alpha1.Cluster, client kubernetes.Client) error {
+	// create namespace on mgmt
+	ns := &v1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cluster.Namespace,
+		},
+	}
+
+	if err := client.Create(ctx, ns); err != nil && !apierrors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, "creating cluster namespace")
+	}
+
+	cluster.ResourceVersion = ""
+	cluster.UID = ""
+
+	// move eksa cluster
+	if err := client.ApplyServerSide(ctx, defaultFieldManager, cluster, kubernetes.ApplyServerSideOptions{ForceOwnership: true}); err != nil {
```

Comment:
can we use `Create` here? it ensures we don't mistakenly update the cluster object after creation in case of a retry

and just check for `apierrors.IsAlreadyExists` if you want to make it idempotent